opm/simulators/linalg/ISTLSolverBda.cpp

```diff
@@ -108,7 +112,8 @@ apply(Vector& rhs,
 #endif
 
         if (numJacobiBlocks_ > 1) {
-            this->copyMatToBlockJac(matrix, *blockJacobiForGPUILU0_);
+            copyThread = std::make_shared<std::thread>([&](){this->copyMatToBlockJac(matrix, *blockJacobiForGPUILU0_);});
+            
             // Const_cast needed since the CUDA stuff overwrites values for better matrix condition..
             bridge_->solve_system(&matrix, blockJacobiForGPUILU0_.get(),
                                   numJacobiBlocks_, rhs, *wellContribs, result);
```

Comment:
Slightly confusing to see the blockJacobiForGPUILU0_ be used, and written to by solve_system, while the copyThread is writing to it. I think it is correct still, even with the replaceZeroDiagonal() in solve_system because either the copy thread or the main thread will write to a certain index in it, but never both... This required some effort to spot so maybe add a comment with why this works to make it more readable.

Reply:
I added a comment, hopefully it is more clear now. Thanks for pointing this out.